Mathlib/Data/ZMod/Basic.lean

```diff
@@ -1321,10 +1356,48 @@ theorem lift_comp_coe : ZMod.lift n f ∘ ((↑) : ℤ → _) = f :=
 theorem lift_comp_castAddHom : (ZMod.lift n f).comp (Int.castAddHom (ZMod n)) = f :=
   AddMonoidHom.ext <| lift_castAddHom _ _
 
+lemma lift_injective {f : {f : ℤ →+ A // f n = 0}} :
+    Injective (lift n f) ↔ ∀ m, f.1 m = 0 → (m : ZMod n) = 0 := by
+  simp only [← AddMonoidHom.ker_eq_bot_iff, eq_bot_iff, SetLike.le_def,
+    ZMod.intCast_surjective.forall, ZMod.lift_coe, AddMonoidHom.mem_ker, AddSubgroup.mem_bot]
+
 end lift
 
 end ZMod
 
+section AddGroup
+variable {α : Type*} [AddGroup α] {n : ℕ}
+
+--TODO: Additivise
+@[simp]
+lemma nsmul_zmod_val_inv_nsmul (hn : (Nat.card α).Coprime n) (a : α) :
+    n • (n⁻¹ : ZMod (Nat.card α)).val • a = a := by
+  rw [← mul_nsmul', ← mod_natCard_nsmul, ← ZMod.val_natCast, Nat.cast_mul,
+    ZMod.mul_val_inv hn.symm, ZMod.val_one_eq_one_mod, mod_natCard_nsmul, one_nsmul]
+
+@[simp]
+lemma zmod_val_inv_nsmul_nsmul (hn : (Nat.card α).Coprime n) (a : α) :
+    (n⁻¹ : ZMod (Nat.card α)).val • n • a = a := by
+  rw [nsmul_left_comm, nsmul_zmod_val_inv_nsmul hn]
+
+end AddGroup
+
+section Group
+variable {α : Type*} [Group α] {n : ℕ}
+
+--TODO: Fix additivisation
```

Comment:
Is this done?

Reply:
No, sadly `to_additive` chokes on `Inv (ZMod n)`. cc @fpvandoorn

Reply:
OK, maybe add a (very) short explanation that this is the problem (of course if you want to add and issue and link it here even better...).